src/api/ai_help.rs

```diff
@@ -420,42 +403,83 @@ pub async fn ai_help(
                     quota: current.map(AIHelpLimit::from_count),
                     created_at,
                 };
-                let tx = log_errors_and_record_response(
-                    &diesel_pool,
-                    history_enabled(&settings),
-                    user.id,
-                    help_ids,
-                )?;
-                let stream = client.chat().create_stream(ai_help_req.req).await.unwrap();
                 let refs = stream::once(async move {
-                    Ok(sse::Event::Data(
+                    Ok::<_, OpenAIError>(sse::Event::Data(
                         sse::Data::new_json(ai_help_meta).map_err(OpenAIError::JSONDeserialize)?,
                     ))
                 });
 
-                Ok(Either::Left(sse::Sse::from_stream(refs.chain(
-                    stream.map_ok(move |res| {
-                        if let Some(ref tx) = tx {
-                            if let Err(e) = tx.send(res.clone()) {
-                                error!("{e}");
-                            }
-                        }
-                        sse::Event::Data(sse::Data::new_json(res).unwrap())
-                    }),
-                ))))
+                Ok(sse::Sse::from_stream(
+                    refs.chain(
+                        stream
+                            .map(Some) // Wrapping response chunks in some.
+                            .chain(stream::once(async move { None })) // Adding a None at the end.
```

Comment:
Is it idiomatic in Rust to add trailing line comments? I would find it more readable if they were each in the line before.

Reply:
Yes for short explainers like this yes.

But if we don't want them then we can change this.